retrievaltool/src/main/java/org/duracloud/retrieval/source/DuraStoreSpecifiedRetrievalSource.java

```diff
@@ -55,6 +68,81 @@ public DuraStoreSpecifiedRetrievalSource(ContentStore store,
+            log.debug("searching for contentIds in space: " + currentRetrievalSpaceId);
+            try {
+                InputStream manifest = contentStore.getManifest(currentRetrievalSpaceId, ManifestFormat.TSV);
+                ManifestFormatter formatter = new ManifestFormatterFactory().create(ManifestFormat.TSV);
+                String header = formatter.getHeader();
+                BufferedReader reader = new BufferedReader(new InputStreamReader(manifest));
+                String line = null;
+                ManifestItem item = null;
+
+                try {
+                    while ((line = reader.readLine()) != null) {
+                        // ignore any whitespace
+                        if (line.trim().length() == 0) {
+                            continue;
+                        }
+
+                        // ignore header line
+                        if (line.equals(header)) {
+                            continue;
+                        }
+
+                        try {
+                            item = formatter.parseLine(line);
+                        } catch (ParseException e) {
+                            throw new IOException(e);
+                        }
+
+                        String contentId = item.getContentId();
+
+                        // search for chunks or chunk manifests for contentId
+                        String tempContentId = contentId;
+                        if (contentId.endsWith(ChunksManifest.manifestSuffix)) {
+                            tempContentId = StringUtils.removeEnd(contentId, ChunksManifest.manifestSuffix);
+                        } else if (contentId.contains(ChunksManifest.chunkSuffix)) {
+                            tempContentId = contentId.substring(0, contentId.indexOf(ChunksManifest.chunkSuffix));
+                        }
+
+                        if (retrievalContentIds.contains(tempContentId)) {
+                            log.debug("found chunked content for contentId in list-file: " + contentId);
+                            retrievalSpaceContentIds.add(contentId);
+                        }
+                    }
+                } catch (IOException ex) {
+                    log.error("Error reading space manifest.");
+                }
+            } catch (ContentStoreException cse) {
+                log.error("Unable to retrieve space manifest. If files-list.txt contains chunked files and the " +
+                          "retrieval fails the local content dir will need to be empty.");
+            }
+        }
+
+        if (!retrievalSpaceContentIds.isEmpty()) {
+            Collections.sort(retrievalSpaceContentIds);
+            log.debug("List of contentIds to retrieve has increased from " + retrievalContentIds.size() + " to " +
+                     retrievalSpaceContentIds.size() + " after chunked content is included.");
+            this.specifiedContentIds = retrievalSpaceContentIds.iterator();

```

Comment:
I'm concerned about the possibility of files in the original retrieval list not being included in this final list if they didn't show up in the manifest (if the file was added very recently, or something is wrong with the manifest). I'd suggest adding for this scenario. If a file was listed in the original file list and it didn't show up in the manifest, I'd be inclined to log that fact then include the file in the final list. It might fail to retrieve if it's a chunked file, but I'd much rather produce an obvious failure rather than silently ignoring one of the files in the list.